test/integration/tracked_watermark_buffer.cc

```diff
@@ -94,5 +135,75 @@ std::pair<uint32_t, uint32_t> TrackedWatermarkBufferFactory::highWatermarkRange(
   return std::make_pair(min_watermark, max_watermark);
 }
 
+bool TrackedWatermarkBufferFactory::waitUntilTotalBufferedExceeds(
+    uint64_t byte_size, std::chrono::milliseconds timeout) {
+  absl::MutexLock lock(&mutex_);
+  auto predicate = [this, byte_size]() ABSL_SHARED_LOCKS_REQUIRED(mutex_) {
+    mutex_.AssertHeld();
+    return total_buffer_size_ >= byte_size;
+  };
+  return mutex_.AwaitWithTimeout(absl::Condition(&predicate), absl::Milliseconds(timeout.count()));
+}
+
+void TrackedWatermarkBufferFactory::removeDanglingAccounts() {
+  auto accounts_it = account_infos_.begin();
+  while (accounts_it != account_infos_.end()) {
+    auto next = std::next(accounts_it);
+
+    // Remove all "dangling" accounts.
```

Comment:
Should you consider accounts with no associated buffers as dangling instead of looking at the ref count?